demo/depth.gdshader

```diff
@@ -0,0 +1,23 @@
+shader_type spatial;
+
+global uniform sampler2DArray environment_depth_texture;
```

Comment:
The PR does not work at all, and I suspect this to be the reason. The texture has a DEPTH_COMPONENT16 format, so I guess it should be used with a shadow sampler. But I can't figure out if and how Godot supports those.